subprojects/docs/src/docs/userguide/java.adoc

```diff
@@ -0,0 +1,810 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[java]]
+== Building Java & JVM projects
+
+Gradle uses a convention-over-configuration approach to building JVM-based projects that borrows several conventions from Apache Maven. In particular, it uses the same directory structure for source files and resources, and it works with Maven-compatible repositories.
+
+We will look at Java projects in detail in this chapter, but most of the topics apply to other supported JVM languages as well, such as https://guides.gradle.org/building-kotlin-jvm-libraries/[Kotlin], <<groovy_plugin,Groovy>> and <<scala_plugin,Scala>>. If you don't have much experience with building JVM-based projects with Gradle, take a look at the <<tutorial_java_projects,Java Quickstart>> first as it will give you a good overview of the basics.
+
+=== Introduction
+
+The simplest build script for a Java project is this:
+
+++++
+<sample id="applyJavaPlugin" dir="userguide/java/basic" title="Applying the Java Plugin">
+    <sourcefile file="build.gradle" snippet="apply-java-plugin"/>
+</sample>
+++++
+
+The properties are optional, but we recommend that you specify them in your projects. The compatibility options mitigate against problems with the project being built with different Java compiler versions, and the version string is important for tracking the progression of the project, as well as being used in archive names.
+
+By applying the Java Plugin, you get a whole host of features:
+
+ * A `compileJava` task that compiles all the Java source files under _src/main/java_
+ * A `compileTestJava` task for source files under _src/test/java_
+ * A `test` task that runs the unit tests from _src/test/java_
+ * A `jar` task that packages the `main` compiled classes and resources from _src/main/resources_ into a single JAR named _<project>-<version>.jar_
+ * A `javadoc` task that generates Javadoc for the `main` classes
+
+This isn't sufficient to build any non-trivial Java project — at the very least, you'll probably have some file dependencies. But it means that your build script only needs the information that is specific to _your project_.
+
+The Java Plugin also integrates the above tasks into the standard <<sec:base_tasks,Base Plugin lifecycle tasks>>:
+
+ * `jar` is attached to `assemble` footnote:[In fact, any artifact added to the `archives` configuration will be built by `assemble`]
+ * `test` is attached to `check`
+
+The rest of the chapter explains the different avenues for customizing the build to your requirements. You will also see later how to adjust the build for libraries, applications, web apps and enterprise apps.
+
+[[sec:java_source_sets]]
+=== Declaring your source code (source sets)
```

Comment:
Instead of using parenthesis we can maybe just say "...with source sets".

Reply:
- [ ] Do this